### ISSUE TITLE
Spyder 5.4.1 not starting with BASH functions in environment

### ISSUE THREAD

**Issue by chhu** [spyder-ide/spyder#20311]
## Problem Description
During start:
```
File "/home/xxx/.conda/envs/spy/lib/python3.10/site-packages/spyder/utils/environ.py", line 63, in get_user_environment_variables
    k, v = kv.split('=', 1)
ValueError: not enough values to unpack (expected 2, got 1)
```

Looking at my env there are multi-line "bash-functions" that crash the split. I cannot remove them easily as bash is my default shell and it recreates them.

Working fix:

spyder/utils/environ.py 
```py
def get_user_environment_variables():
    """
    Get user environment variables from a subprocess.

    Returns
    -------
    env_var : dict
        Key-value pairs of environment variables.
    """
    if os.name == 'nt':
        cmd = "set"
    else:
        cmd = "printenv"
    proc = run_shell_command(cmd)
    stdout, stderr = proc.communicate()
    res = stdout.decode().strip().split(os.linesep)
    env_var = {}
    for kv in res:
        try:
           k, v = kv.split('=', 1)
           env_var[k] = v
        except ValueError:
           continue

    return env_var


```

**Comment by chhu** [comment]
Related to https://github.com/spyder-ide/spyder/pull/20106

Surprised that it is not more urgent?

**Comment by ccordoba12** [comment]
Hey @chhu, thanks for reporting. This is a duplicate of issue #20097 and it'll be fixed in our next version (5.4.2), to be released as soon as possible due to the severity of this bug.
